client/src/components/IconLink.tsx

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright 2024 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useRef } from "react";
+import { Link } from "react-router-dom";
+import { UncontrolledTooltip } from "reactstrap";
+
+interface IconLinkProps {
+  /** An icon to display */
+  icon: React.ReactNode;
+  /** The URL to link to */
+  to: string;
+  /** The text of the tooltip */
+  tooltip: string;
+}
+/**
+ * IconLink
+ * Internal application link that is shown as a font-awesome icon
+ */
+export default function IconLink(props: IconLinkProps) {
+  const ref = useRef(null);
+
+  return (
+    <span>
+      <Link ref={ref} to={props.to}>
+        {props.icon}
+      </Link>
+      <UncontrolledTooltip target={ref}>{props.tooltip}</UncontrolledTooltip>
+    </span>
+  );
+}
```

Comment:
This component can be made more re-usable like so:

```suggestion
import { ReactNode, useRef } from "react";
import { Link, LinkProps } from "react-router-dom-v5-compat";
import { UncontrolledTooltip } from "reactstrap";

type LinkWithTooltipProps = LinkProps & {
  tooltip?: ReactNode;
};

export default function LinkWithTooltip({
  tooltip,
  ...props
}: LinkWithTooltipProps) {
  const ref = useRef<HTMLAnchorElement>(null);

  return (
    <>
      <Link ref={ref} {...props} />
      <UncontrolledTooltip target={ref}>{tooltip}</UncontrolledTooltip>
    </>
  );
}
```

Then to use it:

```tsx
<LinkWithTooltip tooltip="Open workflow" to={url}>
  <Link45deg className={cx("bi", "text-rk-yellow")} />
</LinkWithTooltip>
```